pallets/proofs-dealer/src/lib.rs

```diff
@@ -527,5 +555,15 @@ pub mod pallet {
         fn integrity_test() {
             // TODO: Check that the `CheckpointChallengePeriod` is greater or equal to the largest period of a Provider.
         }
+
+        /// This hook is used to trim down the `ValidProofSubmittersLastBlocks` StorageMap up to the `TargetBlocksOfProofsStorage`.
+        ///
+        /// It runs when the block is being finalized (but before the `on_finalize` hook) and can consume all remaining weight.
+        /// It returns the used weight, so it can be used to calculate the remaining weight for the block for any other
+        /// pallets that have `on_idle` hooks.
+        fn on_idle(n: BlockNumberFor<T>, weight: Weight) -> Weight {
```

Comment:
Add tests for this hook

Reply:
I added a few, I could come up with a few more but they would be mostly redundant. Let me know if there's an specific scenario that you want tested